sgl-router/src/mcp/inventory.rs

```diff
@@ -65,122 +44,57 @@ impl ToolInventory {
     // ============================================================================
 
     /// Get a tool if it exists
-    pub fn get_tool(&self, tool_name: &str) -> Option<(String, Tool)> {
-        self.tools
-            .get(tool_name)
-            .map(|entry| (entry.server_name.clone(), entry.tool.clone()))
+    ///
+    /// Accepts qualified tool name in the format `server_label__tool_name`
+    /// Returns (server_label, server_url, tool)
+    pub fn get_tool(&self, qualified_tool_name: &str) -> Option<(String, String, Tool)> {
+        self.tools.get(qualified_tool_name).map(|entry| {
+            (
+                entry.server_label.clone(),
+                entry.server_url.clone(),
+                entry.tool.clone(),
+            )
+        })
     }
 
     /// Check if tool exists
-    pub fn has_tool(&self, tool_name: &str) -> bool {
-        self.tools.contains_key(tool_name)
+    pub fn has_tool(&self, qualified_tool_name: &str) -> bool {
+        self.tools.contains_key(qualified_tool_name)
     }
 
     /// Insert or update a tool
-    pub fn insert_tool(&self, tool_name: String, server_name: String, tool: Tool) {
-        self.tools
-            .insert(tool_name, CachedTool { server_name, tool });
-    }
-
-    /// Get all tools
-    pub fn list_tools(&self) -> Vec<(String, String, Tool)> {
-        self.tools
-            .iter()
-            .map(|entry| {
-                let (name, cached) = entry.pair();
-                (
-                    name.clone(),
-                    cached.server_name.clone(),
-                    cached.tool.clone(),
-                )
-            })
-            .collect()
-    }
-
-    // ============================================================================
-    // Prompt Methods
-    // ============================================================================
-
-    /// Get a prompt if it exists
-    pub fn get_prompt(&self, prompt_name: &str) -> Option<(String, Prompt)> {
-        self.prompts
-            .get(prompt_name)
-            .map(|entry| (entry.server_name.clone(), entry.prompt.clone()))
-    }
-
-    /// Check if prompt exists
-    pub fn has_prompt(&self, prompt_name: &str) -> bool {
-        self.prompts.contains_key(prompt_name)
-    }
-
-    /// Insert or update a prompt
-    pub fn insert_prompt(&self, prompt_name: String, server_name: String, prompt: Prompt) {
-        self.prompts.insert(
-            prompt_name,
-            CachedPrompt {
-                server_name,
-                prompt,
-            },
-        );
-    }
-
-    /// Get all prompts
-    pub fn list_prompts(&self) -> Vec<(String, String, Prompt)> {
```

Comment:
Can we keep the prompts and resources methods? They may be useful in the future.